### PR TITLE
[Vol2-RL] RL chapter - value-iteration notebook

### PR DESCRIPTION
**Pull request by rasoolfa**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_reinforcement_learning/value-iter.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+#@tab all
+
+
+def value_iteration(einfo, gamma, num_iters):
+    '''
+        gamma: discount factor
+        num_iters: number of iteration
+        env_info: contains MDP and environment related information
+    '''
+    env_desc      = einfo['desc'] # 2D array specifying what each grid item means
```
Comment (cheungdaven) [inline chapter_reinforcement_learning/value-iter.md:108]: This line exceeds the maximum line length 78, you can check the length by enabling the ruler extension. please refer to http://book.d2l.ai/user/code.html for more details.

**Review by cheungdaven — CHANGES_REQUESTED** [review]

File: chapter_reinforcement_learning/value-iter.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+## Action-value function
+
+In implementations, it is often useful to maintain a quantity called the "action value" function which is a closely related quantity to the value function. This is defined to be the average return of a trajectory that begins at $s_0$ but when the action of the first stage is fixed to be $a_0$
+
+$$Q^\pi(s_0, a_0) = r(s_0, a_0) + E_{a_t \sim \pi(s_t)} \Big[ \sum_{t=1}^\infty \gamma^t r(s_t, a_t) \Big],$$
+
+note that the summation inside the expectation is from $t=1,\ldots, \infty$ because the reward of the first stage is fixed in this case. We can also write the principle of dynamic programming in :eqref:`eq_dynamic_programming` using the action-value function as
+
+$$Q^\pi(s, a) = r(s, a) + \gamma \sum_{s' \in \mathcal{S}} P(s' \mid s, a) \sum_{a' \in \mathcal{A}} \pi(a' \mid s')\ Q^\pi(s', a')\text{ for all } s \in \mathcal{S}, a \in \mathcal{A}.$$:eqlabel:`eq_dynamic_programming_q`
```
[the remark this answers is not in the document]
Comment (cheungdaven) [reply chapter_reinforcement_learning/value-iter.md:38]: Sure, feel free to leave the space here, just don’t forget “;”. Thanks.

[the remark this answers is not in the document]
Comment (rasoolfa) [reply chapter_reinforcement_learning/value-iter.md:38]: tnx

File: chapter_reinforcement_learning/value-iter.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+Both the value function and the action-value function depend upon the policy that the robot chooses. We will next think of the "optimal policy" that achieves the maximal average return
+$$\pi^* = \underset{\pi}{\text{argmax}} V^\pi(s_0).$$
+
+Of all possible stochastic policies that the robot could have taken, the optimal policy $\pi^*$  achieves the largest average discounted return for trajectories starting from state $s_0$. Let us denote the value function and the action-value function of the optimal policy as $V^* \equiv V^{\pi^*}$ and $Q^* \equiv Q^{\pi^*}$.
+
+For a deterministic policy where there is only one action that is possible at any given state, we can break down the trajectory of the optimal policy in two stages to observe that
+$$\pi^*(s) = \underset{a \in \mathcal{A}}{\text{argmax}} \Big[ r(s, a) + \gamma \sum_{s' \in \mathcal{S}} P(s' \mid s, a)\ V^*(s') \Big]$$
+
+afterall, the optimal action at stage $s$ is the one that maximizes the sum of reward $r(s, a)$ from the first stage and the average return of the trajectories starting from the next stage $s'$, averaged over all possible next states $s'$.
```
[the remark this answers is not in the document]
Comment (rasoolfa) [reply chapter_reinforcement_learning/value-iter.md:52]: thanks. let me know if this updated version is more clear

**Review by cheungdaven — APPROVED** [review]

**Comment by AnirudhDagar** [comment]
@d2l-bot please rebuild!

**Comment by astonzhang** [comment]
Hi @rasoolfa  Thanks! Could you move this PR to the https://github.com/d2l-ai/d2l-en/tree/vol2-rl branch? We'll go from there soon!

**Comment by astonzhang** [comment]
Closing this PR per offline discussions that new PRs will be posted on d2l-ai:vol2-rl branch

**Pull request closed without merging**